main/lsp/requests/rename.cc

```diff
@@ -133,8 +133,20 @@ class MethodRenamer : public Renamer {
     MethodRenamer(const core::GlobalState &gs, const LSPConfiguration &config, const string oldName,
                   const string newName)
         : Renamer(gs, config, oldName, newName) {
-        if (oldName == "initialize") {
-            invalid = true;
+        const vector<string> invalidNames = {"initialize", "call"};
+        for (auto name : invalidNames) {
+            if (oldName == name) {
+                invalid = true;
+                error = fmt::format("The `{}` method cannot be renamed.", oldName);
+                return;
+            }
+        }
+        // block any method not in /[a-zA-Z0-9_]+/. This blocks operator overloads.
```

Comment:
This will prevent renaming methods like:

```ruby
def has_something?
  true
end

# typically indicates that it mutates something in place
def do_something!

end
```

If this is just for operator overloads, can we just check the first character of `oldName`? I think the first character isn't alnum for those, and I don't think that would block most other renames.

Reply:
Alternatively, you could create a concrete list of operator `NameRefs`.

`Names_gen.h` contains things like `core::Names::unaryPlus()`, `core::Names::squareBrackets()`, etc.

Reply:
If you can pass a `NameRef` in as `oldName`, you can actually compare `NameRef`s (which are always identical if the string matches, and are simply integer comparisons).